### ISSUE TITLE
[Bug]: svg dom parse error about className

### ISSUE THREAD

**Issue by 9kg** [fabricjs/fabric.js#8838]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### Node Version (if applicable)

None

### Link To Reproduction

https://github.com/fabricjs/fabric.js/blob/master/src/parser/selectorMatches.ts

### Steps To Reproduce

When parsing SVG elements, if the class order of an element is "st12 st1", the style of st12 will be lost. In fact, when examining the code, it will be found that st12 hits st1 during the parsing process and leaves a "2". It cannot proceed.

[error code link](https://github.com/fabricjs/fabric.js/blob/master/src/parser/selectorMatches.ts)

### Expected Behavior

Correctly parse the style.

### Actual Behavior

lose style

### Error Message & Stack Trace

```bash
none
```

**Comment by gloriousjob** [comment]
Can you supply an example svg and an actual repro of the problem (a link to source code is not a repro as it doesn't show the effects of the problem)? If you have an idea for a fix, you should create a PR.

**Comment by asturur** [comment]
yes i understand where the issue here is, but indeed a clear reproduction case that can be transformed in a test would be needed.
Probably we need a matcher for the exact class name and not a generic match
